### DISCUSSION TITLE
Dialogic - Out of bounds get index '-1' (on base:Array')

### DISCUSSION THREAD

**Discussion by KimS8989** [dialogic-godot/dialogic#2095] [General]
Hello everyone,

I have been attempting to resolve this issue since this morning without success. In my script, the player is required to click on three different buttons, each initiating a dialogue. So far, there have been no issues with this part. However, when I try to start a new dialogue after all three dialogues have been completed, it doesn't work. My signals seem to be functioning correctly, as I am able to print a test message. The crash occurs only when I add Dialogic.start("intro_sister") instead of print; it then throws an 'Out of bounds get index '-1' (on base: 'Array')' error.

Thank you in advance!

Here is my code:

https://snappify.com/view/545388ec-340d-413c-ba57-6fc903567d74

![image](https://github.com/dialogic-godot/dialogic/assets/144168422/503ad72c-fbe3-4bc3-9b62-350aa124826e)
![image](https://github.com/dialogic-godot/dialogic/assets/144168422/a2cc537c-c0a8-47bd-beca-922fdb17890a)
![image](https://github.com/dialogic-godot/dialogic/assets/144168422/ee348ba8-e2f5-4e89-9fef-2ab3c66eb2e9)

**Comment by CakeVR** [comment]
Do you ensure you call `Dialogic.end_timeline` if you prematurely start a timeline?

**Reply by KimS8989** [reply]
Probably not as I am not sure I understand what you are saying. What code and where I should add it to ensure I call end_timeline?

**Reply by CakeVR** [reply]
If your timeline is not ending normally by reaching the end of the timeline, you should ensure to call `Dialogic.end_timeline()` before starting a new timeline.

**Reply by KimS8989** [reply]
I added Dialogic.end_timeline() and it doesn't work.

<img width="686" alt="Capture d’écran, le 2024-02-17 à 22 00 51" src="https://github.com/dialogic-godot/dialogic/assets/144168422/939f067c-31d6-40e5-9214-8ae6d724bc6f">

**Reply by CakeVR** [reply]
Can you try the unreleased version that sits on the `main` branch?

**Reply by KimS8989** [reply]
Link, please? I want to be sure I download the right one. Thanks for your support!

**Reply by CakeVR** [reply]
Just the default branch (called `main`): https://github.com/dialogic-godot/dialogic

**Reply by KimS8989** [reply]
This is the one I am currently using with my Godot 4.2

**Reply by KimS8989** [reply]
Someone on the Godot forum said the following:

Unfortunately the error occurs inside the Dialogic addon.
Specifically here:

func is_voiced(index:int) -> bool:
	if dialogic.current_timeline_events[index] is DialogicTextEvent:
		if dialogic.current_timeline_events[index-1] is DialogicVoiceEvent:
			return true
	return false
For whatever reason index (or index -1) = -1 and that fires an error in one of the if statements.

**Reply by CakeVR** [reply]
@Jowan-Spooner has fixed this. Try out the latest commit on the `main` branch.

**Reply by notomm** [reply]
I installed the commit in the main branch but I keep on getting this error on every new timeline I try to create. How can I fix this?
